doc/dev/sharp_bits.rst

```diff
@@ -0,0 +1,995 @@
+<https://jax.readthedocs.io/en/latest/notebooks/Common_Gotchas_in_JAX.html>`__.
+
+JAX integration
+---------------
+
+Compiled functions remain JAX compatible, and you can call JAX transformations
+on them, such as ``jax.grad`` and ``jax.vmap``. You can even call ``jax.jit``
+on functions that call qjit-compiled functions:
+
+>>> dev = qml.device("lightning.qubit", wires=2)
+>>> @qjit
+... @qml.qnode(dev)
+... def circuit(x):
+...     qml.RX(x, wires=0)
+...     return qml.expval(qml.PauliZ(0))
+>>> @jax.jit
+... def workflow(y):
+...     return jax.grad(circuit)(jnp.sin(y))
+>>> workflow(0.6)
+Array(-0.53511382, dtype=float64, weak_type=True)
+
+However, a ``jax.jit`` function calling a ``qjit`` function will always result
+in a callback to Python, so will be slower than if the function was purely compiled
+using ``jax.jit`` or ``qjit``.
+
+If you want to compile some functionality that is not currently Catalyst
+compatible, or you want to make use of JAX-supported hardware such as TPUs
+for classical processing, mixing ``jax.jit`` and ``qjit`` will allow this.
+However, if possible, try to always use ``qjit`` to compile your entire
+workflow.
+
+Internal QJIT transformations
+-----------------------------
+
+Inside of a qjit-compiled function, JAX transformations
+(``jax.grad``, ``jax.jacobian``, ``jax.vmap``, etc.)
+can be used **as long as they are not applied to quantum processing**.
+
+>>> @qjit
+... def f(x):
+...     def g(y):
+...         return -jnp.sin(y) ** 2
+...     return jax.grad(g)(x)
+>>> f(0.4)
+array(-0.71735609)
+
+If they are applied to quantum processing, an error will occur:
+
+>>> @qjit
+... def f(x):
+...     @qml.qnode(dev)
+...     def g(y):
+...         qml.RX(y, wires=0)
+...         return qml.expval(qml.PauliX(0))
+...     return jax.grad(lambda y: g(y) ** 2)(x)
+>>> f(0.4)
+NotImplementedError: must override
+
+Instead, only Catalyst transformations will work when applied to hybrid
+quantum-classical processing:
+
+>>> @qjit
+... def f(x):
+...     @qml.qnode(dev)
+...     def g(y):
+...         qml.RX(y, wires=0)
+...         return qml.expval(qml.PauliZ(0))
+...     return grad(lambda y: g(y) ** 2)(x)
+>>> f(0.4)
+array(-0.71735609)
+
+Always use the equivalent Catalyst transformation
+(:func:`catalyst.grad`, :func:`catalyst.jacobian`, :func:`catalyst.vjp`, :func:`catalyst.jvp`)
+inside of a qjit-compiled function.
+
+Inspecting and drawing circuits
+-------------------------------
+
+A useful tool for debugging quantum algorithms is the ability to draw them. Currently,
+:func:`@qjit <~.qjit>` compiled QNodes can be used as input to
+:func:`qml.draw <pennylane.draw>`, with the following caveats:
+
+- :func:`qml.draw <pennylane.draw>` call must occur outside the :func:`@qjit <.qjit>`
+
+- If used, the :func:`@qjit <.qjit>` decorator must be placed directly on top of the QNode (that is, you cannot draw a hybrid `qjit` compiled function that _contains_ a QNode).
```

Comment:
```suggestion
- The `qml.draw()` function will only accept plain QNodes as input, _or_ QNodes that have been qjit-compiled. It will not accept arbitrary hybrid functions (that may contain QNodes).
```

Reply:
@dime10 rewrote this to make it clearer

Reply:
Awesome, this one is very clear 💯